pkg/logs/input/journald/tailer_util.go

```diff
@@ -0,0 +1,58 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+package journald
+
+import (
+	"fmt"
+	"math/rand"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/util/cache"
+)
+
+const (
+	imageTagKey             = "short_image"
+	tagSeparator            = ":"
+	defaultShortNameValue   = "docker"
+	baseCacheExpiration     = 20 * time.Minute
+	expirationSpreadSeconds = 120
+)
+
+// Get docker image short from dockerId and list of tags
+// In case of miss in cache, we parse from tag list.
+// We could use the util/Docker.Inspect + ImageID resolution or Regexp on config.Image
+// But this way is simple and probably fast enough (~300ns for 100 items) on miss
+func getDockerImageShortName(containerID string, tags []string) (string, bool) {
+	cacheKey := getImageCacheKey(containerID)
+	cacheContent, hit := cache.Cache.Get(cacheKey)
+	if hit {
+		return cacheContent.(string), true
+	}
+
+	// Cache miss
+	var shortName string
+	for _, tag := range tags {
+		if strings.HasPrefix(tag, imageTagKey) {
+			tagParts := strings.Split(tag, tagSeparator)
+			if len(tagParts) == 2 {
+				shortName = tagParts[1]
+				break
+			}
+		}
+	}
+
+	if shortName != "" {
+		cache.Cache.Set(cacheKey, shortName, baseCacheExpiration+time.Duration(rand.Intn(expirationSpreadSeconds))*time.Second)
```

Comment:
why do we want to add an extra expiration time ? worth adding a comment to explain please

Reply:
if tags was a map, we would not need this cache I think, WDYT ? Also, do we really want to cache this ? what's the cardinality of tags ? Is it gonna be a huge gain ?

Reply:
Correct, if tag is map, then nothing to do. Earlier comment answers the why. Expiration is just to make sure we purge at some point when a docker is stopped/deleted. By default cache has a 5mins retention I believe, which is a bit low in this case as the docker_id <> image_name is never going to change.

Indeed, I asked and cardinality could be in ~100s in worst cases, which I benchmarked to be around ~300ns on my laptop. As the code is called for every single entry in journald, it could be causing high cpu usage on the agent if we have a heavy logger, so better avoid it.

I'll add a comment to explain this.

Reply:
Sounds good thanks, resolving the conversation.